config/load_test.go

```diff
@@ -587,6 +587,22 @@ func TestLoad(t *testing.T) {
 			},
 		},
 		{
+			args: []string{"-ui.certfile", "cert_value"},
+			cfg: func(cfg *Config) *Config {
+				cfg.UI.CertFile = "cert_value"
+				cfg.Registry.Consul.CheckScheme = "https"
+				return cfg
+			},
+		},
+		{
+			args: []string{"-ui.keyfile", "key_value"},
+			cfg: func(cfg *Config) *Config {
+				cfg.UI.KeyFile = "key_value"
+				cfg.Registry.Consul.CheckScheme = "http"
```

Comment:
The way this is currently coded it is, in that if you just specify a keyFile it will ignore it and still server http, as you need to specify a certFile to get https. I don't think it would work with just a private key.